src/quantuminspire/qiskit/qi_job.py

```diff
@@ -30,9 +30,9 @@ def __init__(self, backend: Any, job_id: str, api: QuantumInspireAPI, qobj: Opti
         """
         self._api: QuantumInspireAPI = api
         super().__init__(backend, job_id)
-        self.experiments: Optional[List[QobjExperiment]] = None
-        self._status: str = JobStatus.INITIALIZING
-        self._qobj: Optional[Qobj] = qobj
+        self.experiments: Optional[List[QasmQobjExperiment]] = None
+        self._status: JobStatus = JobStatus.INITIALIZING
```

Comment:
Actually it is an enumeration type (Jobstatus). Each item is a string but the type is enumeration. When python would be strongly typed something like self._status = "random string" should not be possible, because it's not an enumeration of JobStatus. When we make the type of self._status string (str) self._status = "random string" is possible, which is wrong.